File: src/main/java/me/testaccount666/serversystem/commands/executables/stack/CommandStack.java
package me.testaccount666.serversystem.commands.executables.stack;

import me.testaccount666.serversystem.commands.ServerSystemCommand;
import me.testaccount666.serversystem.commands.executables.AbstractServerSystemCommand;
import me.testaccount666.serversystem.managers.PermissionManager;
import me.testaccount666.serversystem.userdata.ConsoleUser;
import me.testaccount666.serversystem.userdata.User;
import org.bukkit.command.Command;
import org.bukkit.entity.Player;

import static me.testaccount666.serversystem.utils.MessageBuilder.command;
import static me.testaccount666.serversystem.utils.MessageBuilder.general;

@ServerSystemCommand(name = "stack")
public class CommandStack extends AbstractServerSystemCommand {

    @Override
    public boolean hasCommandAccess(Player player, Command command) {
        return PermissionManager.hasCommandPermission(player, "Stack.Use", false);
    }

    @Override
    public void execute(User commandSender, Command command, String label, String... arguments) {
        if (!checkBasePermission(commandSender, "Stack.Use")) return;
        if (commandSender instanceof ConsoleUser) {
            general("NotPlayer", commandSender).build();
            return;
        }

        var itemInHand = commandSender.getPlayer().getInventory().getItemInMainHand();
        if (itemInHand.getType().isAir()) {
            command("Stack.NoItemInHand", commandSender).build();
            return;
        }

        itemInHand.setAmount(itemInHand.getMaxStackSize());
        command("Stack.Success", commandSender).build();
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/commands/executables/stack/CommandStack.kt
package me.testaccount666.serversystem.commands.executables.stack

import me.testaccount666.paperktx.extensions.get
import me.testaccount666.paperktx.extensions.isAir
import me.testaccount666.serversystem.commands.ServerSystemCommand
import me.testaccount666.serversystem.commands.executables.AbstractServerSystemCommand
import me.testaccount666.serversystem.extensions.commandMsg
import me.testaccount666.serversystem.userdata.User
import org.bukkit.command.Command
import org.bukkit.inventory.EquipmentSlot

@ServerSystemCommand("stack")
class CommandStack : AbstractServerSystemCommand() {
    override fun getUsagePermission(command: Command) = "Stack.Use"
    override fun getSyntaxPath(command: Command?) = "Generic"

    override fun execute(commandSender: User, command: Command, label: String, vararg arguments: String) {
        if (!isPlayer(commandSender)) return

        val itemInHand = commandSender.getPlayer()?.inventory[EquipmentSlot.HAND].takeUnless { it.isAir() } ?: run {
            commandSender.commandMsg("Stack.NoItemInHand")
            return
        }

        itemInHand.amount = itemInHand.maxStackSize
        commandSender.commandMsg("Stack.Success")
    }
}
